ray_casting/List[] list
